jobsdb/jobsdb.go

```diff
@@ -1831,61 +1831,77 @@ func (jd *HandleT) postMigrateHandleDS(migrateFrom []dataSetT) error {
 Next set of functions are for reading/writing jobs and job_status for
 a given dataset. The names should be self explainatory
 */
-func (jd *HandleT) storeJobsDS(ds dataSetT, copyID bool, jobList []*JobT) error { //When fixing callers make sure error is handled with assertError
+func (jd *HandleT) storeJobsDS(ds dataSetT, jobList []*JobT) error { //When fixing callers make sure error is handled with assertError
 	queryStat := jd.storeTimerStat("store_jobs")
 	queryStat.Start()
 	defer queryStat.End()
 
-	txn, err := jd.dbHandle.Begin()
-	if err != nil {
-		return err
-	}
-
 	// Always clear cache even in case of an error,
 	// since we are not sure about the state of the db
-	defer func() {
-		customValParamMap := make(map[string]map[string]map[string]struct{})
-		var workspaces []string //for bursting old cache
-		for _, job := range jobList {
-			if !misc.ContainsString(workspaces, job.WorkspaceId) {
-				workspaces = append(workspaces, job.WorkspaceId)
-			}
-			jd.populateCustomValParamMap(customValParamMap, job.CustomVal, job.Parameters, job.WorkspaceId)
-		}
+	defer jd.clearCache(ds, jobList)
```

Comment:
@lvrach to consider for the future, since caching appears to be an aspect (horizontal concern). Instead of injecting caching logic in each and every method of `jobsdb.HandleT` we could use the decorator pattern (embedded struct) and handle all caching concerns on a different layer.
[This example](https://fabianlindfors.se/blog/decorators-in-go-using-embedded-structs/) also deals with caching 🤣

Reply:
Totally agree with that, I have proposed a jobsdb decorator a while back. Be prepared caused it might need some effort to introduce.